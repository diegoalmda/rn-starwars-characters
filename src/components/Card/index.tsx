import { FC } from "react";
import { View, Text } from "react-native";
import { styles } from './styles';

interface CardProps {
  characterInfo: {
    name: string;
    height: string;
    mass: string;
  }
}

export const Card:FC<CardProps> = ({ characterInfo }: CardProps) => {
  return (
    <View style={styles.container} >
      <Text style={styles.textName}>{characterInfo.name}</Text>
      <View>
        <Text style={styles.itemText}>Height: {characterInfo.height}</Text>
        <Text style={styles.itemText}>Mass: {characterInfo.mass}</Text>
      </View>
    </View>
  );
}