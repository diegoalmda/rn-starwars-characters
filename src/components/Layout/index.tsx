import { FC, ReactNode } from 'react';
import { StatusBar } from 'expo-status-bar';
import { ImageBackground, Platform, SafeAreaView, StyleSheet } from 'react-native';

import backgroundImage from '../../assets/background.png';

interface LayoutProps {
  children: ReactNode
}

export const Layout:FC = ({ children }: LayoutProps) => {
  return (
    <ImageBackground 
      style={styles.container}
      source={backgroundImage}
      defaultSource={backgroundImage}
      imageStyle={styles.cover}
    >
      <StatusBar
        style='light'
        backgroundColor='transparent'
        translucent
      />
      <SafeAreaView style={styles.AndroidSafeArea}>
        { children }
      </SafeAreaView>
    </ImageBackground>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#000',
    alignItems: 'center',
    justifyContent: 'center',
  },
  cover: {
    width: '100%',
    height: '100%',
    position: 'absolute',
    resizeMode: 'cover',
    opacity: 0.5,
  },
  AndroidSafeArea: {
    flex: 1, 
    marginVertical: Platform.OS === 'android' ? 30 : 0,
    width: '100%',
    paddingHorizontal: 10,
  }
});