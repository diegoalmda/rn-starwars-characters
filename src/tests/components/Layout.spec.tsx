import React from 'react';
import { Text } from "react-native";
import { Layout } from "../../components/Layout";
import { cleanup, render } from "@testing-library/react-native";
 
afterEach(cleanup);

describe('Layout Component', () => {
  
  it('renders Child component', () => {
    const helloWorldText = <Text>Hello World!</Text>;
 
    const {toJSON, getByText} = render(<Layout children={helloWorldText} />);
 
    const foundHelloWorldText = getByText('Hello World!');
 
    expect(foundHelloWorldText.props.children).toEqual('Hello World!');
    expect(toJSON()).toMatchSnapshot();
  });
});