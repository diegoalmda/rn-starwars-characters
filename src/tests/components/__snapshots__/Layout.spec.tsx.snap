// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component renders Child component 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    defaultSource={1}
    source={1}
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": undefined,
          "width": undefined,
        },
        {
          "height": "100%",
          "opacity": 0.5,
          "position": "absolute",
          "resizeMode": "cover",
          "width": "100%",
        },
      ]
    }
  />
  <RCTSafeAreaView
    style={
      {
        "flex": 1,
        "marginVertical": 0,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text>
      Hello World!
    </Text>
  </RCTSafeAreaView>
</View>
`;
